src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationService.kt

```diff
@@ -627,11 +644,65 @@ class AdjudicationService(
       mapOf(
         "adjudicationNumber" to adjudicationNumber.toString(),
         "hearingId" to hearingId.toString(),
-        "resultSequence" to "1",
+        "chargeSequence" to chargeSequence.toString(),
       ),
       null,
     )
   }
+
+  @Audit
+  fun createHearingResultAward(
+    adjudicationNumber: Long,
+    requests: CreateHearingResultAwardRequests,
+  ): CreateHearingResultAwardResponse {
+    // DPS does not associate award with result (migrated or synchronised data)
+
+    val party = adjudicationIncidentPartyRepository.findByAdjudicationNumber(adjudicationNumber)
+      ?: throw NotFoundException("Adjudication party with adjudication number $adjudicationNumber not found")
+
+    val offenderBookId = party.offenderBooking!!.bookingId // the adjudication party will always have a booking Id
+
+    val sanctionSeq =
+      adjudicationHearingResultAwardRepository.getNextSanctionSequence(offenderBookId = offenderBookId)
+
+    // latest result should always be target result
+    val hearingResult =
+      adjudicationHearingResultRepository.findFirstOrNullByIncidentChargeOrderById_resultSequenceDesc(
+        party.charges.single(),
```

Comment:
I think here, don't we need to latest result for a charge? In other words I suspect we also need to charge sequence for migrated adjudications where there are multiple charges

Reply:
yes - will add